src/Player/Equipment/Weapon.cpp

```diff
@@ -2,44 +2,71 @@
 // Created by adven on 2024/3/24.
 //
 #include "pch_extream.h"
-
 #include "Player/Equipment/Weapon.h"
 
 Weapon::Weapon() {
-    //    const auto ItemImage = std::make_shared<Util::Image>(m_ImagePathItem);
-    const auto ItemSize = ToolBoxs::CountImagePixel(m_ImagePathItem, 1, 2);
 
-    const auto ItemImage = std::make_shared<SpriteSheet>(
-        m_ImagePathItem, ItemSize, std::vector<std::size_t>{0}, false, 100, true, 100
-    );
-    const auto WindowImage = std::make_shared<Util::Image>(m_ImagePathWindow);
+    rendererWindow();
 
-    m_Window->SetDrawable(WindowImage);
-    m_Item->SetDrawable(ItemImage);
+    setWeaponType(WeaponEnum::Type::DAGGER);
```

Comment:
![image](https://github.com/NTUT-FUCK-PTSD/Fuck-PTSD/assets/25154413/cd8b8506-7619-4fb8-8937-61d123c4b502)

![image](https://github.com/NTUT-FUCK-PTSD/Fuck-PTSD/assets/25154413/33cdb227-fc57-4742-bfd6-ea3660dbe74e)

Reply:
Do not call overridable methods in constructors
Reference: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2214